bundles/CoreBundle/Controller/PublicServicesController.php

```diff
@@ -56,6 +61,37 @@ public function thumbnailAction(Request $request)
             if ($asset->getPath() === ('/' . $prefix)) {
                 // we need to check the path as well, this is important in the case you have restricted the public access to
                 // assets via rewrite rules
+
+                if ($asset instanceof Asset\Video && $request->get('type') == 'video') {
```

Comment:
Isn't this already handled here https://github.com/pimcore/pimcore/pull/13311/files#diff-80c761bc7a7fb7998e853a73bfc7ddb1a083639cc26b42c314ffa845633d94acR132?

Reply:
That's for `image-thumb` of a video, in other words the `cover` image, while what i am adding is serving `video-thumb` when request type matches "video" in "video-thumb"

Reply:
Ok...then how about we incorporate this into already existing if else checks instead of repeating the code for returning the response?